packages/fs.go

```diff
@@ -28,31 +29,52 @@ type PackageFileSystem interface {
 // ExtractedPackageFileSystem provides utils to access files in an extracted package.
 type ExtractedPackageFileSystem struct {
 	path string
+	root fs.FS
 }
 
 func NewExtractedPackageFileSystem(p *Package) (*ExtractedPackageFileSystem, error) {
-	return &ExtractedPackageFileSystem{path: p.BasePath}, nil
+	return &ExtractedPackageFileSystem{path: p.BasePath, root: os.DirFS(p.BasePath)}, nil
 }
 
-func (fs *ExtractedPackageFileSystem) Stat(name string) (os.FileInfo, error) {
-	return os.Stat(filepath.Join(fs.path, name))
+func (fsys *ExtractedPackageFileSystem) Stat(name string) (os.FileInfo, error) {
+	name = strings.TrimPrefix(name, "/")
+	return fs.Stat(fsys.root, name)
 }
 
-func (fs *ExtractedPackageFileSystem) Open(name string) (PackageFile, error) {
-	return os.Open(filepath.Join(fs.path, name))
-}
-
-func (fs *ExtractedPackageFileSystem) Glob(pattern string) (matches []string, err error) {
-	matches, err = filepath.Glob(filepath.Join(fs.path, pattern))
+func (fsys *ExtractedPackageFileSystem) Open(name string) (PackageFile, error) {
+	name = strings.TrimPrefix(name, "/")
+	f, err := fsys.root.Open(name)
 	if err != nil {
-		return
+		return nil, err
 	}
-	for i := range matches {
-		match, err := filepath.Rel(fs.path, matches[i])
+	// f is a plain os.File expected to implement the PackageFile interface.
+	pf, ok := f.(PackageFile)
+	if !ok {
+		defer f.Close()
+		return nil, fmt.Errorf("file does not implement PackageFile interface: %q", filepath.Join(fsys.path, filepath.FromSlash(name)))
+	}
+	return pf, nil
+}
+
+func (fsys *ExtractedPackageFileSystem) Glob(pattern string) (matches []string, err error) {
+	e := fs.WalkDir(fsys.root, ".", func(p string, d fs.DirEntry, err error) error {
```

Comment:
Nit. Error variables in go are usually called `err`.
```suggestion
	err := fs.WalkDir(fsys.root, ".", func(p string, d fs.DirEntry, err error) error {
```
Or you can probably reuse the one already defined in the function.
```suggestion
	err = fs.WalkDir(fsys.root, ".", func(p string, d fs.DirEntry, err error) error {
```
In cases where you need to distinguish between different errors in the same scope, you can use `Err` as suffix.
```suggestion
	walkErr := fs.WalkDir(fsys.root, ".", func(p string, d fs.DirEntry, err error) error {
```